.circleci/config.yml

```diff
@@ -0,0 +1,89 @@
+version: 2.1  # The version of CircleCI platform being used
+executors:
+  python-docker:
+    working_directory: ~/repo
+    docker:
+      - image: cimg/python:3.6
+
+
+jobs:  # The jobs level contains a collection of arbitrarily named children
+  run-tests:
+    executor: python-docker
+    steps:  # The steps collection is an ordered list of run directives
+      - checkout  # "checkout" checks out the source code to "working_directory"
+      - restore_cache:
```

Comment:
to makes more efficient use of the cache this should be:
```yml
- restore_cache:
    keys:
      - pip-deps-v1-{{ arch }}-{{ checksum "setup.py" }}-{{ checksum "constraints.txt" }}
      - pip-deps-v1-{{ arch }}-{{ checksum "setup.py" }}
      - pip-deps-v1-{{ arch }}
      - pip-deps-v1
```

Reply:
why? We don't do this in other repos

Reply:
Do we have a consensus on pip-deps-v1-{{ arch }}-{{ checksum "setup.py" }}-{{ checksum "constraints.txt" }}? I can put that in but @anishk25 do you believe I should not?

Reply:
@migel can you advise on this? getting close to wrapping this up.

Reply:
Got it- @anishk25 while for this it makes no sense, apparently its an optimization on prefixes. Ill throw it in and if you guys ever want we can remove it.

Reply:
"Incorrect type. Expected "string".yaml-schema: JSON schema for CircleCI configuration files(0)" when I switched this out.

Reply:
fallback keys: https://circleci.com/blog/six-optimization-tips-for-your-config/ @anishk25 --- I certainly doubt this will matter on this build but definitely worth knowing about.